src/globe/Globe.js

```diff
@@ -659,6 +659,38 @@ define([
             return this.elevationModel.elevationsForGrid(sector, numLat, numLon, targetResolution, result);
         };
 
+        /**
+         * Computes a new set of locations translated from a specified location to a new location for this globe.
+         *
+         * @param {Location} oldLocation The original reference location.
+         * @param {Location} newLocation The new reference location.
+         * @param {Location[]} locations The locations to translate.
+         *
+         * @return {Location[]} The translated locations.
+         */
+        Globe.prototype.computeShiftedLocations = function(oldLocation, newLocation, locations) {
+            var newLocations = [];
+
+            var oldPoint = this.computePointFromLocation(oldLocation.latitude, oldLocation.longitude,
+                new Vec3(0, 0, 0));
+            var newPoint = this.computePointFromLocation(newLocation.latitude, newLocation.longitude,
+                new Vec3(0, 0, 0));
+            var delta = newPoint.subtract(oldPoint);
+
+            for(var i =0; i < locations.length; i++)
+            {
+                var point = this.computePointFromLocation(locations[i].latitude, locations[i].longitude,
+                    new Vec3(0, 0, 0));
+                point = point.add(delta);
+                var newPos = new WorldWind.Position(0, 0, 0);
+                this.computePositionFromPoint(point[0], point[1], point[2], newPos);
+
+                newLocations.push(newPos);
+            }
+
+            return newLocations;
```

Comment:
This actually returns an array of positions, not locations. That's what the Java implementation does as well, but I believe we should correct this here. @pdavidc what do you think?

Reply:
Yes, I agree that returning locations would be best. In this context position's altitude is at best misleading, and at worst exposes the app to an error prone interpretation of the results. 

Regarding how this is accomplished, there's an important subtlety. I'd like to avoid adding a `computeLocationFromPoint` to Globe (on a related note, `computePointFromLocation` needs to be deprecated sometime in the future). These duplicate functions appear to be a nice convenience, but they have several problems: they hide necessary complexity (a point does not actually convert to a location and vice-versa), they complicate Globe's interface, and they expand the test suite needed to validate its Globe's correctness.

Given that, we'll need to use `computePositionFromPoint` as we do now, but convert each result to a location before its stored in newLocations. Memory overhead can be eliminated by declaring a scratch Position before the loop, in a manner identical to @AkeluX's comment about the Vec3.

Reply:
@claudiaifrim please transform the returned list of `Position` objects into a list of `Location` objects as proposed by Dave here.